File: codes/go/chapter05/answer_A32.go
package main

import "fmt"

func main() {
	var N, A, B int
	fmt.Scan(&N, &A, &B)

	dp := make([]bool, N+1)

	for i := 0; i <= N; i++ {
		if i >= A && !dp[i-A] {
			dp[i] = true
		} else if i >= B && !dp[i-B] {
			dp[i] = true
		} else {
			dp[i] = false
		}
	}

	if dp[N] {
		fmt.Println("First")
	} else {
		fmt.Println("Second")
	}
}

File: codes/ruby/2/chapter05/answer_A32.rb
n, a, b = gets.split.map(&:to_i)

dp = Array.new(n + 1)

(0..n).each do |i|
  dp[i] = if i >= a && dp[i - a] == false
            true
          elsif i >= b && dp[i - b] == false
            true
          else
            false
          end
end

if dp[n] == true
  puts 'First'
else
  puts 'Second'
end
